### PR TITLE
add Doctrine Embedded support

### PR DESCRIPTION
**Pull request by vvh-empora**

### CODE REVIEW

File: src/Matmar10/Money/Entity/Money.php [hunk 1]
```diff
@@ -7,9 +7,8 @@
 use JMS\Serializer\Annotation\ReadOnly;
 use JMS\Serializer\Annotation\SerializedName;
 use JMS\Serializer\Annotation\Type;
-use Matmar10\Money\Entity\CurrencyInterface;
-use Matmar10\Money\Entity\MoneyInterface;
```
Comment (matmar10) [inline src/Matmar10/Money/Entity/Money.php:11]: Why is this line removed? The class still needs to implement MoneyInterface

File: src/Matmar10/Money/Entity/Money.php [hunk 2]
```diff
@@ -7,9 +7,8 @@
 use JMS\Serializer\Annotation\ReadOnly;
 use JMS\Serializer\Annotation\SerializedName;
 use JMS\Serializer\Annotation\Type;
-use Matmar10\Money\Entity\CurrencyInterface;
```
Comment (matmar10) [inline src/Matmar10/Money/Entity/Money.php:10]: Why is this removed? The class still uses CurrencyInterface

File: src/Matmar10/Money/Entity/Money.php [hunk 3]
```diff
@@ -79,17 +112,17 @@ public function getCurrency()
 
     public function getScale()
     {
-        return $this->scale;
+        return $this->scale ?: $this->scale = bcpow(10, $this->currency->getPrecision(), 0);
```
Comment (matmar10) [inline src/Matmar10/Money/Entity/Money.php]: What's the goal of moving the assignment here? Now their is a condition being evaluated any time someone calls getScale()

File: src/Matmar10/Money/Entity/Money.php [hunk 4]
```diff
@@ -62,14 +64,45 @@ class Money implements MoneyInterface
...
+    public static function create(CurrencyInterface $currency, $amount = null)
+    {
+        return new static($currency, $amount);
+    }
+
+    /**
+     * @param CurrencyInterface     $currency
+     * @param int|float|string|null $amount
+     */
+    public function __construct(CurrencyInterface $currency, $amount = null)
```
Comment (matmar10) [inline src/Matmar10/Money/Entity/Money.php:82]: This seems like a great change. Can you add a test case for the various types that are now allowed to be fed into the constructor?

File: src/Matmar10/Money/Entity/Money.php (hunk 1 above)
Comment (vvh-empora) [inline src/Matmar10/Money/Entity/Money.php:11]: we are in the same namespace, we didnt need it...

File: src/Matmar10/Money/Entity/Money.php (hunk 2 above)
Comment (vvh-empora) [inline src/Matmar10/Money/Entity/Money.php:10]: we are in the same namespace, we didnt need it...

File: src/Matmar10/Money/Entity/Money.php (hunk 3 above)
Comment (vvh-empora) [inline src/Matmar10/Money/Entity/Money.php]: this is a trade of to use this class as embedded property. there is no event or callback to set the scale after load this entity from database as embedded style :(

File: src/Matmar10/Money/Entity/Money.php (hunk 4 above)
Comment (vvh-empora) [inline src/Matmar10/Money/Entity/Money.php:82]: sure

Comment (matmar10) [inline src/Matmar10/Money/Entity/Money.php:82]: :+1: :smile:

File: src/Matmar10/Money/Entity/Money.php (hunk 3 above)
Comment (matmar10) [inline src/Matmar10/Money/Entity/Money.php]: Ah OK, that makes sense. Can you change it to be less clever?

```
    if (!$this->scale) {
        $this->scale = bcpow(10, $this->currency->getPrecision(), 0);
    }

    return $this->scale;
```

The line as it's written now is harder to groc.

File: src/Matmar10/Money/Entity/Money.php (hunk 2 above)
Comment (matmar10) [inline src/Matmar10/Money/Entity/Money.php:10]: Ok cool. :+1:

File: src/Matmar10/Money/Entity/Money.php (hunk 1 above)
Comment (matmar10) [inline src/Matmar10/Money/Entity/Money.php:11]: :+1:

File: src/Matmar10/Money/Entity/Money.php (hunk 3 above)
Comment (vvh-empora) [inline src/Matmar10/Money/Entity/Money.php]: hmmm... you dont like my "do it in one line" action style ? ;)

**Comment by vvh-empora** [comment]
push :+1:
